src/metrics/coranking/script.R

```diff
@@ -30,8 +30,11 @@ if (any(is.na(X_emb))) {
   # TODO: computing a square distance matrix is problematic for large datasets!
   # TODO: should we use a different distance metric for the high_dim?
   # TODO: or should we subset to the HVG?
-  dist_highdim <- coRanking:::euclidean(as.matrix(high_dim))
-  dist_emb <- coRanking:::euclidean(as.matrix(X_emb))
+  message("Compute high-dimensional distances")
+  dist_highdim <- proxyC::dist(high_dim, method = "euclidean")
+  diag(dist_highdim) <- 0 # Force diagonal to be 0
```

Comment:
You can force the diagonal to be 0 by setting `diag = TRUE`.

Granted, maybe this should be the default when x == y...

```suggestion
  dist_highdim <- proxyC::dist(high_dim, method = "euclidean", diag = TRUE, drop0 = TRUE)
```